playbooks/roles/download-and-verify/tasks/main.yml

```diff
@@ -1,41 +1,50 @@
 ---
-- name: "{{ project_signer }} signs the {{ project_name }} downloads. Import the correct GPG key."
-  command: gpg --keyserver {{ gpg_key_server_address }} --recv-keys {{ project_signing_key }}
-  register: gpg_recv_keys_result
-  changed_when: False
-  until: gpg_recv_keys_result | success
-  retries: 10
-  delay: 5
 
-- name: Register the GPG Key fingerprint output
-  command: gpg --fingerprint {{ project_signing_key }}
-  register: retrieved_fingerprint
-  changed_when: False
+# Import the GPG playbook vars
+- include_vars: "../../gpg/vars/main.yml"
 
-- name: Make sure the retrieved fingerprint perfectly matches the expected fingerprint
-  assert: { that: "project_expected_fingerprint in retrieved_fingerprint.stdout" }
-
-- name: Download the {{ project_name }} files and signatures
+- name: "Download the {{ project_name }} files"
   get_url:
-    url: "{{ item }}"
-    dest: "{{ project_download_location }}"
+    url: "{{ project_download_baseurl }}/{{ item.file }}"
+    dest: "{{ project_download_location }}/{{ item.file }}"
     owner: www-data
     group: www-data
     mode: 0644
-  with_items: "{{ project_download_urls }}"
+  with_items: "{{ project_download_files }}"
 
-- name: If necessary, rename signature files to match the GPG default of '.asc'
-  shell: for file in *.{{ signature_extension }}; do mv $file ${file%.*}.asc; done
-  args:
-    chdir: "{{ project_download_location }}"
-  when: signature_extension != "asc"
+- name: "Download the {{ project_name }} signatures"
+  get_url:
+    url: "{{ project_download_baseurl }}/{{ item.sig }}"
+    dest: "{{ project_download_location }}/{{ item.sig }}"
+    owner: www-data
+    group: www-data
+    mode: 0644
+  with_items: "{{ project_download_files }}"
 
-- name: Register GPG signature verification results
-  shell: for file in *.asc; do gpg --verify $file; done
-  args:
-    chdir: "{{ project_download_location }}"
+- name: "Verify the {{ project_name }} download signatures with the Streisand GPG keyring"
+  command: "gpgv2 --keyring {{ streisand_gpg_keyring }} {{ project_download_location }}/{{ item.sig }} {{ project_download_location }}/{{ item.file }}"
   register: gpg_verification_results
-  changed_when: False
+  with_items: "{{ project_download_files }}"
+
+- name: "Verify the {{ project_name }} download signature checks all passed"
+  assert:
+    that:
+      - "not item.failed"
+      - "item.rc == 0"
+    msg: "Verifying {{ item.item.file }} GPG signature failed!!"
+  with_items: "{{ gpg_verification_results.results }}"
 
-- name: Make sure the {{ project_name }} files all passed GPG signature verification
-  assert: { that: "'BAD signature' not in gpg_verification_results.stderr" }
+- name: "Verify the {{ project_name }} download signatures were from the correct keys"
```

Comment:
This is kinda long too. Your call on the matter of taste; personally, I'd want to keep the diagnostics in until everything else is done....